packages/relay/src/lib/eth.ts

```diff
@@ -1403,11 +1403,35 @@ export class EthImpl implements Eth {
     const transactionBuffer = Buffer.from(EthImpl.prune0x(transaction), 'hex');
 
     let txSubmitted = false;
+    let contractExecuteResponse;
+
     try {
-      const contractExecuteResponse = await this.hapiService
+      contractExecuteResponse = await this.hapiService
         .getSDKClient()
         .submitEthereumTransaction(transactionBuffer, EthImpl.ethSendRawTransaction, requestIdPrefix);
       txSubmitted = true;
+    } catch (e: any) {
+      const isJsonRpcError = e instanceof JsonRpcError;
+      const isSDKClientError = e instanceof SDKClientError;
+      const isAllowedSDKClientError = isSDKClientError && !(e.isConnectionDropped() || e.isTimeoutExceeded());
+
+      if (isJsonRpcError || isAllowedSDKClientError) {
+        return this.sendRawTransactionErrorHandler(e, transaction, transactionBuffer, txSubmitted, requestIdPrefix);
+      }
+
+      if (isSDKClientError) {
+        this.logger.warn(`${requestIdPrefix} SDK Client has probably timed out, trying again with a new instance...`);
+        this.hapiService.decrementErrorCounter(e.statusCode);
+      }
+    }
+
+    try {
+      if (!txSubmitted) {
+        contractExecuteResponse = await this.hapiService
+          .getSDKClient()
+          .submitEthereumTransaction(transactionBuffer, EthImpl.ethSendRawTransaction, requestIdPrefix);
+        txSubmitted = true;
+      }
```

Comment:
hmm, I think it's because using async functions as callbacks is kinda tricky and we have to do sth similar to: 
```typescript
  return new Promise(resolve => {
    return doIt().then(result => resolve(result));
  });
```

Reply:
Works now, thanks 👍